Revert the following:

commit bc40c180447a813feea283c7bcb531113a37738e
Author: Nico Weber <thakis@chromium.org>
Date:   Fri May 15 11:23:44 2026 -0400

    Implement Math.{acos,asin}() using LLVM's libm
    
    After a fix [1] to my benchmark to not mostly pass in values that
    are out of domain, this comes with a fairly small perf hit, in
    return for full accuracy.
    
    Before:
    
    ====================================================================
    Function   Max ULP    Mean ULP    % CR    % FR  Accuracy   Ops/sec
    --------------------------------------------------------------------
    Math.acos     1.00      0.1296   87.0%  100.0%      88.5    164.1M
    Math.asin     1.00      0.0285   97.1%  100.0%      97.2    174.6M
    
    After:
    
    Math.acos     0.00      0.0000  100.0%  100.0%     100.0    130.4M
    Math.asin     0.00      0.0000  100.0%  100.0%     100.0    149.2M
    
    (Passing in values outside the domain of [-1, 1] now produces nan
    quite a bit slower. Before, we hit ~500-600 Ops/sec, now we hit
    ~170 Ops/sec for that use case.
    `#define LIBC_MATH (LIBC_MATH_NO_ERRNO | LIBC_MATH_NO_EXCEPT)` gets
    it to ~310 Ops / sec, so we'll likely want to set that at some point
    as there isn't really a reason not to do it. But even then, the
    out-of-domain case is quite a bit slower. In practice, this should
    hopefully be rare, though.)
    
    1: https://github.com/nico/js-math-bench/commit/807684d98
    
    Bug: 510499297
    Change-Id: I7985d2fac072186ab5486173553b53613f8755bf
    Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/7849784
    Reviewed-by: Leszek Swirski <leszeks@chromium.org>
    Commit-Queue: Nico Weber <thakis@chromium.org>
    Auto-Submit: Nico Weber <thakis@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#107355}

--- a/v8/src/base/ieee754.cc
+++ b/v8/src/base/ieee754.cc
@@ -95,7 +95,90 @@ namespace ieee754 {
     (d) = base::bit_cast<double>(bits);          \
   } while (false)
 
-double acos(double x) { return LIBC_NAMESPACE::shared::acos(x); }
+/* acos(x)
+ * Method :
+ *      acos(x)  = pi/2 - asin(x)
+ *      acos(-x) = pi/2 + asin(x)
+ * For |x|<=0.5
+ *      acos(x) = pi/2 - (x + x*x^2*R(x^2))     (see asin.c)
+ * For x>0.5
+ *      acos(x) = pi/2 - (pi/2 - 2asin(sqrt((1-x)/2)))
+ *              = 2asin(sqrt((1-x)/2))
+ *              = 2s + 2s*z*R(z)        ...z=(1-x)/2, s=sqrt(z)
+ *              = 2f + (2c + 2s*z*R(z))
+ *     where f=hi part of s, and c = (z-f*f)/(s+f) is the correction term
+ *     for f so that f+c ~ sqrt(z).
+ * For x<-0.5
+ *      acos(x) = pi - 2asin(sqrt((1-|x|)/2))
+ *              = pi - 0.5*(s+s*z*R(z)), where z=(1-|x|)/2,s=sqrt(z)
+ *
+ * Special cases:
+ *      if x is NaN, return x itself;
+ *      if |x|>1, return NaN with invalid signal.
+ *
+ * Function needed: sqrt
+ */
+double acos(double x) {
+  static const double
+      one = 1.00000000000000000000e+00,     /* 0x3FF00000, 0x00000000 */
+      pi = 3.14159265358979311600e+00,      /* 0x400921FB, 0x54442D18 */
+      pio2_hi = 1.57079632679489655800e+00, /* 0x3FF921FB, 0x54442D18 */
+      pio2_lo = 6.12323399573676603587e-17, /* 0x3C91A626, 0x33145C07 */
+      pS0 = 1.66666666666666657415e-01,     /* 0x3FC55555, 0x55555555 */
+      pS1 = -3.25565818622400915405e-01,    /* 0xBFD4D612, 0x03EB6F7D */
+      pS2 = 2.01212532134862925881e-01,     /* 0x3FC9C155, 0x0E884455 */
+      pS3 = -4.00555345006794114027e-02,    /* 0xBFA48228, 0xB5688F3B */
+      pS4 = 7.91534994289814532176e-04,     /* 0x3F49EFE0, 0x7501B288 */
+      pS5 = 3.47933107596021167570e-05,     /* 0x3F023DE1, 0x0DFDF709 */
+      qS1 = -2.40339491173441421878e+00,    /* 0xC0033A27, 0x1C8A2D4B */
+      qS2 = 2.02094576023350569471e+00,     /* 0x40002AE5, 0x9C598AC8 */
+      qS3 = -6.88283971605453293030e-01,    /* 0xBFE6066C, 0x1B8D0159 */
+      qS4 = 7.70381505559019352791e-02;     /* 0x3FB3B8C5, 0xB12E9282 */
+
+  double z, p, q, r, w, s, c, df;
+  int32_t hx, ix;
+  GET_HIGH_WORD(hx, x);
+  ix = hx & 0x7FFFFFFF;
+  if (ix >= 0x3FF00000) { /* |x| >= 1 */
+    uint32_t lx;
+    GET_LOW_WORD(lx, x);
+    if (((ix - 0x3FF00000) | lx) == 0) { /* |x|==1 */
+      if (hx > 0) {
+        return 0.0; /* acos(1) = 0  */
+      } else {
+        return pi + 2.0 * pio2_lo; /* acos(-1)= pi */
+      }
+    }
+    return std::numeric_limits<double>::signaling_NaN();  // acos(|x|>1) is NaN
+  }
+  if (ix < 0x3FE00000) {                            /* |x| < 0.5 */
+    if (ix <= 0x3C600000) return pio2_hi + pio2_lo; /*if|x|<2**-57*/
+    z = x * x;
+    p = z * (pS0 + z * (pS1 + z * (pS2 + z * (pS3 + z * (pS4 + z * pS5)))));
+    q = one + z * (qS1 + z * (qS2 + z * (qS3 + z * qS4)));
+    r = p / q;
+    return pio2_hi - (x - (pio2_lo - x * r));
+  } else if (hx < 0) { /* x < -0.5 */
+    z = (one + x) * 0.5;
+    p = z * (pS0 + z * (pS1 + z * (pS2 + z * (pS3 + z * (pS4 + z * pS5)))));
+    q = one + z * (qS1 + z * (qS2 + z * (qS3 + z * qS4)));
+    s = sqrt(z);
+    r = p / q;
+    w = r * s - pio2_lo;
+    return pi - 2.0 * (s + w);
+  } else { /* x > 0.5 */
+    z = (one - x) * 0.5;
+    s = sqrt(z);
+    df = s;
+    SET_LOW_WORD(df, 0);
+    c = (z - df * df) / (s + df);
+    p = z * (pS0 + z * (pS1 + z * (pS2 + z * (pS3 + z * (pS4 + z * pS5)))));
+    q = one + z * (qS1 + z * (qS2 + z * (qS3 + z * qS4)));
+    r = p / q;
+    w = r * s + c;
+    return 2.0 * (df + w);
+  }
+}
 
 /* acosh(x)
  * Method :
@@ -137,8 +220,101 @@ double acosh(double x) {
   }
 }
 
-double asin(double x) { return LIBC_NAMESPACE::shared::asin(x); }
+/* asin(x)
+ * Method :
+ *      Since  asin(x) = x + x^3/6 + x^5*3/40 + x^7*15/336 + ...
+ *      we approximate asin(x) on [0,0.5] by
+ *              asin(x) = x + x*x^2*R(x^2)
+ *      where
+ *              R(x^2) is a rational approximation of (asin(x)-x)/x^3
+ *      and its remez error is bounded by
+ *              |(asin(x)-x)/x^3 - R(x^2)| < 2^(-58.75)
+ *
+ *      For x in [0.5,1]
+ *              asin(x) = pi/2-2*asin(sqrt((1-x)/2))
+ *      Let y = (1-x), z = y/2, s := sqrt(z), and pio2_hi+pio2_lo=pi/2;
+ *      then for x>0.98
+ *              asin(x) = pi/2 - 2*(s+s*z*R(z))
+ *                      = pio2_hi - (2*(s+s*z*R(z)) - pio2_lo)
+ *      For x<=0.98, let pio4_hi = pio2_hi/2, then
+ *              f = hi part of s;
+ *              c = sqrt(z) - f = (z-f*f)/(s+f)         ...f+c=sqrt(z)
+ *      and
+ *              asin(x) = pi/2 - 2*(s+s*z*R(z))
+ *                      = pio4_hi+(pio4-2s)-(2s*z*R(z)-pio2_lo)
+ *                      = pio4_hi+(pio4-2f)-(2s*z*R(z)-(pio2_lo+2c))
+ *
+ * Special cases:
+ *      if x is NaN, return x itself;
+ *      if |x|>1, return NaN with invalid signal.
+ */
+double asin(double x) {
+  static const double
+      one = 1.00000000000000000000e+00, /* 0x3FF00000, 0x00000000 */
+      huge = 1.000e+300,
+      pio2_hi = 1.57079632679489655800e+00, /* 0x3FF921FB, 0x54442D18 */
+      pio2_lo = 6.12323399573676603587e-17, /* 0x3C91A626, 0x33145C07 */
+      pio4_hi = 7.85398163397448278999e-01, /* 0x3FE921FB, 0x54442D18 */
+                                            /* coefficient for R(x^2) */
+      pS0 = 1.66666666666666657415e-01,     /* 0x3FC55555, 0x55555555 */
+      pS1 = -3.25565818622400915405e-01,    /* 0xBFD4D612, 0x03EB6F7D */
+      pS2 = 2.01212532134862925881e-01,     /* 0x3FC9C155, 0x0E884455 */
+      pS3 = -4.00555345006794114027e-02,    /* 0xBFA48228, 0xB5688F3B */
+      pS4 = 7.91534994289814532176e-04,     /* 0x3F49EFE0, 0x7501B288 */
+      pS5 = 3.47933107596021167570e-05,     /* 0x3F023DE1, 0x0DFDF709 */
+      qS1 = -2.40339491173441421878e+00,    /* 0xC0033A27, 0x1C8A2D4B */
+      qS2 = 2.02094576023350569471e+00,     /* 0x40002AE5, 0x9C598AC8 */
+      qS3 = -6.88283971605453293030e-01,    /* 0xBFE6066C, 0x1B8D0159 */
+      qS4 = 7.70381505559019352791e-02;     /* 0x3FB3B8C5, 0xB12E9282 */
+
+  double t, w, p, q, c, r, s;
+  int32_t hx, ix;
 
+  t = 0;
+  GET_HIGH_WORD(hx, x);
+  ix = hx & 0x7FFFFFFF;
+  if (ix >= 0x3FF00000) { /* |x|>= 1 */
+    uint32_t lx;
+    GET_LOW_WORD(lx, x);
+    if (((ix - 0x3FF00000) | lx) == 0) { /* asin(1)=+-pi/2 with inexact */
+      return x * pio2_hi + x * pio2_lo;
+    }
+    return std::numeric_limits<double>::signaling_NaN();  // asin(|x|>1) is NaN
+  } else if (ix < 0x3FE00000) {     /* |x|<0.5 */
+    if (ix < 0x3E400000) {          /* if |x| < 2**-27 */
+      if (huge + x > one) return x; /* return x with inexact if x!=0*/
+    } else {
+      t = x * x;
+    }
+    p = t * (pS0 + t * (pS1 + t * (pS2 + t * (pS3 + t * (pS4 + t * pS5)))));
+    q = one + t * (qS1 + t * (qS2 + t * (qS3 + t * qS4)));
+    w = p / q;
+    return x + x * w;
+  }
+  /* 1> |x|>= 0.5 */
+  w = one - fabs(x);
+  t = w * 0.5;
+  p = t * (pS0 + t * (pS1 + t * (pS2 + t * (pS3 + t * (pS4 + t * pS5)))));
+  q = one + t * (qS1 + t * (qS2 + t * (qS3 + t * qS4)));
+  s = sqrt(t);
+  if (ix >= 0x3FEF3333) { /* if |x| > 0.975 */
+    w = p / q;
+    t = pio2_hi - (2.0 * (s + s * w) - pio2_lo);
+  } else {
+    w = s;
+    SET_LOW_WORD(w, 0);
+    c = (t - w * w) / (s + w);
+    r = p / q;
+    p = 2.0 * s * r - (pio2_lo - 2.0 * c);
+    q = pio4_hi - 2.0 * w;
+    t = pio4_hi - (p - q);
+  }
+  if (hx > 0) {
+    return t;
+  } else {
+    return -t;
+  }
+}
 /* asinh(x)
  * Method :
  *      Based on
